src/main/java/nextstep/service/ReservationService.java

```diff
@@ -0,0 +1,87 @@
+package nextstep.service;
+
+import auth.domain.persist.UserDetails;
+import auth.support.AuthorizationException;
+import nextstep.domain.dto.request.ReservationRequest;
+import nextstep.domain.dto.response.ReservationResponse;
+import nextstep.domain.persist.Member;
+import nextstep.domain.persist.Reservation;
+import nextstep.domain.persist.Schedule;
+import nextstep.domain.persist.Theme;
+import nextstep.repository.MemberDao;
+import nextstep.repository.ReservationDao;
+import nextstep.repository.ScheduleDao;
+import nextstep.repository.ThemeDao;
+import nextstep.support.exception.api.*;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class ReservationService {
+    private final ReservationDao reservationDao;
+    private final ThemeDao themeDao;
+    private final ScheduleDao scheduleDao;
+    private final MemberDao memberDao;
+
+    public ReservationService(ReservationDao reservationDao, ThemeDao themeDao, ScheduleDao scheduleDao, MemberDao memberDao) {
+        this.reservationDao = reservationDao;
+        this.themeDao = themeDao;
+        this.scheduleDao = scheduleDao;
+        this.memberDao = memberDao;
+    }
+
+    @Transactional
+    public Long create(UserDetails userDetails, ReservationRequest reservationRequest) {
+        if (userDetails == null) {
+            throw new AuthorizationException();
+        }
+        Schedule schedule = scheduleDao.findById(reservationRequest.getScheduleId());
+        if (schedule == null) {
+            throw new NoSuchScheduleException();
+        }
+
+        List<Reservation> reservation = reservationDao.findByScheduleId(schedule.getId());
+        if (!reservation.isEmpty()) {
+            throw new DuplicateReservationException();
+        }
+
+        Reservation newReservation = new Reservation(
+                schedule,
+                new Member(userDetails)
+        );
+
+        return reservationDao.save(newReservation);
+    }
+
+    @Transactional(readOnly = true)
+    public List<Reservation> findAllByThemeIdAndDate(Long themeId, String date) {
+        Theme theme = themeDao.findById(themeId);
+        if (theme == null) {
+            throw new NoSuchThemeException();
+        }
+
+        return reservationDao.findAllByThemeIdAndDate(themeId, date);
+    }
+
+    @Transactional(readOnly = true)
+    public List<ReservationResponse> findAllByUserId(Long id) {
```

Comment:
함수명을 보니 id가 userId를 의미하는거 같아요 
이름을 더 구체적으로 지어주시면 코드 이해하기에 더 도움이 될거같아요~! 

```suggestion
    public List<ReservationResponse> findAllByUserId(Long userId) {
```